lib/vocabularies/discriminator/index.ts

```diff
@@ -62,10 +63,24 @@ const def: CodeKeywordDefinition = {
       const topRequired = hasRequired(parentSchema)
       let tagRequired = true
       for (let i = 0; i < oneOf.length; i++) {
-        const sch = oneOf[i]
-        const propSch = sch.properties?.[tagName]
-        if (typeof propSch != "object") {
-          throw new Error(`discriminator: oneOf schemas must have "properties/${tagName}"`)
+        let sch = oneOf[i]
+        let propSch
+        if (sch?.["$ref"] && Object.keys(sch).length === 1) {
```

Comment:
There is some function that tests whether schema has validation keywords other than ref (ignoring e.g. title) - I will update it later.

Reply:
ah - this is `schemaHasRulesButRef(sch, self.RULES)` see in validate/index